app/views/companies/_companies_list.html.haml

```diff
@@ -30,7 +30,7 @@
         %tr.company
           %td{ style: 'white-space: nowrap;' }
             = company.business_categories.distinct.order(:name).each.map{|bc| bc.name}.join('<br/>').html_safe
-          %td= link_to company.name, company
+          %td= link_to company.name.empty? ? t('name_missing') : company.name, company
```

Comment:
I created a few "non-complete" companies in the DB via FactoryBot, e.g.
```
FactoryBot.create(:company, name: nil)
FactoryBot.create(:company, name: nil, company_number: '9999999999')
```
Then, I saw this error in the index view:

<img width="1017" alt="screen shot 2018-02-27 at 5 43 11 pm" src="https://user-images.githubusercontent.com/9968213/36759626-0494b2aa-1be6-11e8-8e9b-564572a2f6a9.png">

The problem is that `company.name`, when realized, returns `nil` since the name is not defined.

You can fix this by using `blank?` instead of `empty?`, which can handle being sent to a `nil` object:
```
%td= link_to company.name.blank? .....
```

Reply:
http://api.rubyonrails.org/classes/Object.html#method-i-blank-3F

Reply:
And, I know this is a bit unfair, since we only generally create a company with a string for the name - empty or otherwise.  However, we _can_ create companies with `nil` name since we do not validate that attribute, so we should try to protect against this in the code.